src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicNativeMethods.cs

```diff
@@ -0,0 +1,594 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    /// <summary>
+    /// Contains all native delegates and structs that are used with MsQuic.
+    /// </summary>
+    internal static unsafe class MsQuicNativeMethods
+    {
+        internal const string dllName = "msquic";
+
+#pragma warning disable BCL0015 // Disable Pinvoke analyzer errors for now
+        [DllImport(dllName)]
+        internal static extern int MsQuicOpen(int version, out NativeApi* registration);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct NativeApi
```

Comment:
The single entry-point that returns an array of pointers to call looks odd for a C library. It just makes the interop more painful than what it needs to be. And makes it impossible to get rid of unused code when the library is linked statically (important for mobile apps).

Reply:
You have marked this as resolved without any comments, but I do not see it addressed in the delta either.

Reply:
This isn't a question I can answer. @nibanks ?

Reply:
I don't understand the question.

Reply:
The question is why you have to go through the extra hoop to get the pointers to actual method to call. Why not export the C APIs directly?

Reply:
This is how we version the API. We have a different function table per API version. BTW, the API is not statically linked. It's a dynamic library.

Reply:
IMHO, it is very Windows-centric design. The de-facto standards for popular cross-platforms open-source libraries are different.

Reply:
Perhaps, but not many cross platform APIs also have to work in Windows kernel mode and support Windows updates.

Reply:
You can have a simple C API for the public consumption, and keep the Windows kernel/inbox specific extra complexity to be Windows kernel/inbox only.

Reply:
It's all the same API. I have no plans to change this design at this time.